staking/programs/profile/src/lib.rs

```diff
@@ -0,0 +1,74 @@
+//! Profile program
+//!
+//! This program allows a Solana user to map their Solana address to their addresses on other chains
+
+#![allow(clippy::result_large_err)]
+use anchor_lang::prelude::*;
+
+declare_id!("prfmVhiQTN5Spgoxa8uZJba35V1s7XXReqbBiqPDWeJ");
+
+#[program]
+pub mod profile {
+    use super::*;
+
+    pub fn update_identity(ctx: Context<UpdateIdentity>, identity: Identity) -> Result<()> {
+        ctx.accounts.identity_account.identity = identity;
+        Ok(())
+    }
+}
+
+#[derive(Accounts)]
+#[instruction(identity: Identity)]
+pub struct UpdateIdentity<'info> {
+    #[account(mut)]
+    payer:            Signer<'info>,
+    #[account(init_if_needed, payer = payer, space = identity.size(), seeds = [&[identity.to_u8()], payer.key().as_ref()], bump ) ]
+    identity_account: Account<'info, IdentityAccount>,
+    system_program:   Program<'info, System>,
+}
+
+#[account]
+pub struct IdentityAccount {
+    identity: Identity,
+}
+
+#[derive(AnchorDeserialize, AnchorSerialize, Clone)]
+pub enum Identity {
+    Evm { pubkey: Option<[u8; 20]> },
+}
+
+impl Identity {
+    fn to_u8(&self) -> u8 {
```

Comment:
maybe `ecosystem_id_as_u8` ? 

`to_u8` suggests that the whole thing can be a `u8`